### PR TITLE
Fix issues reported by user - JediBees - RES-692

### PR DESCRIPTION
**Pull request by danielamihalache**
Reported issues that were fixed:
- addTo(String email, String name) doesn't seem to work
- HTTP url is used for sending
- IOException is caught, constructor is package protected

### CODE REVIEW

File: Sendgrid.java [hunk 1]
```diff
@@ -527,11 +557,23 @@ public void send() throws IOException, JSONException {
...
+					}
+	                requestParams.append("=");
+	                try {
+						requestParams.append(URLEncoder.encode(value, "UTF-8"));
+					} catch (UnsupportedEncodingException e) {
+						message = "UnsupportedEncodingException - " + e.getMessage();
+					}
+	                requestParams.append("&");
+                  }
+         }
```
Comment (laur-craciun) [inline Sendgrid.java]: please fix the indentation in this method and also in setBcc

File: Sendgrid.java [hunk 2]
```diff
@@ -72,8 +73,13 @@ public Sendgrid setTo(String email) {
      * @return           The SendGrid object.
      */
     public Sendgrid addTo(String email, String name) {
-        String toAddress = (name.length() > 0) ? name + "<" + email + ">" : email;
-        this.to_list.add(toAddress);
+        if (name.length() > 0){
+            this.to_name_list.add(name);
```
Comment (laur-craciun) [inline Sendgrid.java]: use function addToName(name) here

File: Sendgrid.java [hunk 3]
```diff
@@ -72,8 +73,13 @@ public Sendgrid setTo(String email) {
...
+        } else {
+            this.to_name_list.add("");
```
Comment (laur-craciun) [inline Sendgrid.java]: use function addToName("") here

File: Sendgrid.java (hunk 1 above)
Comment (danielamihalache) [inline Sendgrid.java]: done

File: Sendgrid.java (hunk 2 above)
Comment (danielamihalache) [inline Sendgrid.java]: done

File: Sendgrid.java (hunk 3 above)
Comment (danielamihalache) [inline Sendgrid.java]: done

File: Sendgrid.java [hunk 4]
```diff
@@ -59,6 +60,7 @@
     public Sendgrid setTo(String email) {
         this.to_list = new ArrayList<String>();
         this.to_list.add(email);
+        this.to_name_list.add("");
```
Comment (laur-craciun) [inline Sendgrid.java]: I think you should use function addTo(email) intead of :
  this.to_list.add(email);
  this.to_name_list.add("");

Comment (danielamihalache) [inline Sendgrid.java]: done

**Pull request merged**